addons/pos_sale_loyalty/static/src/js/models.js

```diff
@@ -0,0 +1,21 @@
+/** @odoo-module alias=pos_sale_loyalty.models **/
+
+
+import { Orderline } from 'point_of_sale.models';
+import Registries from 'point_of_sale.Registries';
+
+export const PosSaleLoyaltyOrderline = (Orderline) => class PosSaleLoyaltyOrderline extends Orderline {
+    //@override
+    setQuantityFromSOL(saleOrderLine) {
```

Comment:
I don't think you should try to guess if a line is a reward based on a type or price, can't you just check if it has a reward_id?